core/services/collections/Collection.php

```diff
@@ -38,6 +47,37 @@ class Collection extends \SplObjectStorage implements CollectionInterface {
 	  */
 	 public function __construct( $collection_interface ) {
 		 $this->setCollectionInterface( $collection_interface );
+	     $this->setCollectionIdentifier();
+	 }
+
+
+     /**
+      * @return string
+      */
+     public function collectionIdentifier()
+     {
+         return $this->collection_identifier;
+     }
+
+
+
+	 /**
+      * creates a very readable unique 9 character identifier like:  CF2-532-DAC
+      * and appends it to the non-qualified class name, ex: ThingCollection-CF2-532-DAC
+      *
+	  * @return void
+	  */
+	 protected function setCollectionIdentifier() {
+         // hash a few collection details
```

Comment:
I'm curious to know the reasoning for this beyond just creating a human friendly identifier.  In other words, **why** are you creating a human friendly identifier?

Also, again, this is something that I think should be done in a separate branch as an atomic commit rather than lumped in with a potentially long living branch.

Reply:
> yaaa... I could go through the trouble of creating another issue and branch for every little thing added to core, but that just adds extra stuff on to everyone's plates, and since this is not a high priority need, and this class rarely changes, I don't mind it being in this branch.

Just to be clear that's not what I'm saying here (that we have to create another issue and branch for every little thing added to core).  Put another way, what I mean is if the change we are making is something that is not directly related to this branch and touches a part of the codebase that might get changed elsewhere in another branch, it's probably _safer_ to do it as an atomic commit in a branch off of master and get it merged _back_ to master right away.  I don't think that adds extra load to us and actually potentially _helps_ prevent future issues because it reduces the chance of merge conflicts.  Also, the more atomic the commit, the quicker it gets through code review and the easier it is to verify via a quick test.

If you don't think this particular file will get modified in any other branch or that you could use your change elsewhere, then sure just leave it here.